src/ir_Amcor.cpp

```diff
@@ -144,49 +145,45 @@ void IRAmcorAc::on(void) { setPower(true); }
 void IRAmcorAc::off(void) { setPower(false); }
 
 void IRAmcorAc::setPower(const bool on) {
-  remote_state[kAmcorPowerByte] &= ~kAmcorPowerMask;
-  remote_state[kAmcorPowerByte] |= (on ? kAmcorPowerOn : kAmcorPowerOff);
+  setBits(&remote_state[kAmcorPowerByte], kAmcorPowerOffset, kAmcorPowerSize,
+          on ? kAmcorPowerOn : kAmcorPowerOff);
 }
 
 bool IRAmcorAc::getPower(void) {
-  return ((remote_state[kAmcorPowerByte] & kAmcorPowerMask) == kAmcorPowerOn);
+  return GETBITS8(remote_state[kAmcorPowerByte], kAmcorPowerOffset,
+                  kAmcorPowerSize) == kAmcorPowerOn;
 }
 
 // Set the temp in deg C
 void IRAmcorAc::setTemp(const uint8_t degrees) {
   uint8_t temp = std::max(kAmcorMinTemp, degrees);
   temp = std::min(kAmcorMaxTemp, temp);
-
-  temp <<= 1;
-  remote_state[kAmcorTempByte] &= ~kAmcorTempMask;
-  remote_state[kAmcorTempByte] |= temp;
+  setBits(&remote_state[kAmcorTempByte], kAmcorTempOffset, kAmcorTempSize,
+          temp);
 }
 
 uint8_t IRAmcorAc::getTemp(void) {
-  return (remote_state[kAmcorTempByte] & kAmcorTempMask) >> 1;
+  return GETBITS8(remote_state[kAmcorTempByte], kAmcorTempOffset,
+                  kAmcorTempSize);
 }
 
 // Maximum Cooling or Hearing
 void IRAmcorAc::setMax(const bool on) {
   if (on) {
     switch (getMode()) {
-      case kAmcorCool:
-        setTemp(kAmcorMinTemp);
-        break;
-      case kAmcorHeat:
-        setTemp(kAmcorMaxTemp);
-        break;
-      default:  // Not allowed in all other operating modes.
-        return;
+      case kAmcorCool: setTemp(kAmcorMinTemp); break;
+      case kAmcorHeat: setTemp(kAmcorMaxTemp); break;
+      // Not allowed in all other operating modes.
+      default: return;
     }
-    remote_state[kAmcorSpecialByte] |= kAmcorMaxMask;
-  } else {
-    remote_state[kAmcorSpecialByte] &= ~kAmcorMaxMask;
   }
+  setBits(&remote_state[kAmcorSpecialByte], kAmcorMaxOffset, kAmcorMaxSize,
+          on ? kAmcorMax : 0);
 }
 
 bool IRAmcorAc::getMax(void) {
-  return ((remote_state[kAmcorSpecialByte] & kAmcorMaxMask) == kAmcorMaxMask);
+  return GETBITS8(remote_state[kAmcorSpecialByte], kAmcorMaxOffset,
+                  kAmcorMaxSize) == kAmcorMax;
 }
```

Comment:
Are all masks changed, or just specific ones?
I find the the new way of using offsets much harder to understand compared to using masks - this might lead to it being harder for others to understand the code, and contribute. - I do clearly see the benefits in other places.
It might however be better to have functions that works with mask directly, instead of introducing offsets.

Reply:
Almost all masks have been changed. The exceptions at this stage are masks like `0b00100100` where the mask isn't contiguous.
Personally, I preferred simple bit masks. but a lot of people had trouble understanding them. When refactoring the code, I found a few examples of either poor masks that I had done, or ones that were wrong.
I often get them wrong when writing the code but people don't see that as I write unit tests to make sure things behave the way I expect, not the way I've initially written the code. :-)

You can still use the old ways of doing it.
I'll probably add some routines to allow arbitrary masks to this PR.